Don't clear NuGet caches at all

dougbu opened the pull request:
- `$use_global_nuget_cache` and `$useGlobalNuGetCache` have default values in our pipelines
- therefore, cache is always within `$(System.DefaultWorkingDirectory)` and should not exist at job start

dougbu committed:
Always clear NuGet caches
- not sure why this was ever Windows-specific
- also remove `NuGetToolInstaller@1` task since tool is installed by default
  - should avoid `ERR:certificate has expired` errors
  - we don't care what version is installed

dougbu commented:
/fyi @MattGal @jonfortescue @ulisesh @alexperovich since this PR may mean the `ERR:certificate has expired` errors should affect us less, especially as I backport this to servicing branches

dougbu commented on .azure/pipelines/jobs/default-build.yml:
```diff
@@ -178,14 +178,11 @@ jobs:
-      - task: NuGetToolInstaller@1
-      - task: NuGetCommand@2
-        displayName: 'Clear NuGet caches'
-        condition: succeeded()
```
Note `succeeded()` is the default `condition`

MattGal commented on .azure/pipelines/jobs/default-build.yml:
```diff
@@ -178,14 +178,11 @@ jobs:
-        inputs:
-          command: custom
-          arguments: 'locals all -clear'
+    - task: NuGetCommand@2
```
Are you sure this works?  Looking at the virtual environments repo, there's no nuget CLI installed on hosted linux agents and AFAIK not on ours either.

It may make more sense to just do `dotnet nuget locals all -clear` once a .NET Core SDK is installed.

dougbu replied:
I see lots of failures in this PR but the errors are `Unable to locate executable file: 'mono'. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also check the file mode to verify the file is executable.` _after_ the tool is found. That looks like a problem w/ the configuration of _some_ agents or in _some_ Docker images. Many of the Linux and macOS jobs look fine.

<details>
  <summary>Complete log</summary>

  ``` text
  Found tool in cache: NuGet 4.1.0 x64
  Found tool in cache: NuGet 4.1.0 x64
  Resolved from tool cache: 4.1.0
  Using version: 4.1.0
  Found tool in cache: NuGet 4.1.0 x64
  SYSTEMVSSCONNECTION exists true
  SYSTEMVSSCONNECTION exists true
  Detected NuGet version 4.1.0.2450 / 4.1.0
  SYSTEMVSSCONNECTION exists true
  ##[error]Error: Unable to locate executable file: 'mono'. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also check the file mode to verify the file is executable.
  ```

</summary>

dougbu replied:
> It may make more sense to just do `dotnet nuget locals all -clear` once a .NET Core SDK is installed.

Is `dotnet` included in all container and agent configurations❔ If not, the SDK won't be installed until the build step that does the rest of our build.

dougbu replied:
I'd forgotten about the `$env:NUGET_PACKAGES` overrides in the tools.* scripts. We shouldn't need any of this.

dougbu committed:
Instead don't clear NuGet caches at all
- `$use_global_nuget_cache` and `$useGlobalNuGetCache` have default values in our pipelines
- therefore, cache is always within `$(System.DefaultWorkingDirectory)` and should not exist at job start
